src/shmem_atomic.h

```diff
@@ -77,52 +77,47 @@ static inline
 void
 shmem_spinlock_unlock(shmem_spinlock_t *lock)
 {
-    COMPILER_FENCE();
-    lock->exit++;
+    __atomic_fetch_add(&lock->exit, 1, __ATOMIC_RELEASE);
```

Comment:
Although `fetch` is not necessary, should we use `add_fetch` instead?

Reply:
Yeah, the ``__atomic`` builtins don't provide a non-fetching atomic add operation.  I'm not sure if fetch-add or add-fetch is faster.  🤷‍♂